flask_oauthlib/client.py

```diff
@@ -621,15 +621,15 @@ def handle_oauth1_response(self):
             )
         return data
 
-    def handle_oauth2_response(self):
+    def handle_oauth2_response(self, args):
         """Handles an oauth2 authorization response."""
 
         client = self.make_client()
         remote_args = {
-            'code': request.args.get('code'),
+            'code': args.get('code'),
             'client_secret': self.consumer_secret,
-            'redirect_uri': session.get('%s_oauthredir' % self.name)
         }
+        remote_args['redirect_uri'] = session.get('%s_oauthredir' % self.name)
```

Comment:
I cannot quite catch why this change is necessary. Could you, please, shed some light on it?